config/lib/mongoose.js

```diff
@@ -29,6 +29,8 @@ module.exports.connect = function (cb) {
       console.log(err);
     } else {
 
+      mongoose.Promise = config.db.promise;
```

Comment:
I'm sort of indifferent on whether or not to make this configurable. Personally, I like things to be as configurable as possible. However, in this case I can't think of any use case where having a different promise library to be used between, say, a dev env & test env.

If we do go with adding this into the environment configuration, I agree with @shanavas786 that having it in the default config is sufficient. This wouldn't stop someone from adding it to the others if they have a need to do so.